### PR TITLE
SCL-71: File Upload Limits and Filetype Limiting

### PR DESCRIPTION
**Pull request by sandaiiyahh**
- add validation for large file uploads (> 1MB)
- add validation for invalid image file (couldn't test mobile on web locally since on web, it was already not allowed but hopefully this now works too on mobile 😅)

Screen Shot 2023-05-10 at 10 17 34 PM

### CODE REVIEW

File: app/src/components/task-completion/TaskUpload.tsx [hunk 1]
```diff
@@ -67,15 +67,27 @@ type TaskUploadProps = {
...
 }
 
 export default function TaskUpload(props: TaskUploadProps) {
-  const { imageFileSrc, setImage, setImageFileSrc } = props;
+  const { imageFileSrc, setImage, setImageFileSrc, setInvalidImageMessage } = props;
 
   const onImageChange = (event: any) => {
+    console.log('event.target.files[0]',  event.target.files[0])
     if (event.target.files && event.target.files[0]) {
+      const allowedTypes = ['image/jpeg', 'image/jpg', 'image/png']
```
Comment (jacoblb64) [inline app/src/components/task-completion/TaskUpload.tsx:79]: might have to add `heic` for iphones?

File: app/src/components/task-completion/TaskUpload.tsx [hunk 2]
```diff
@@ -67,15 +67,27 @@ type TaskUploadProps = {
...
+      if (!allowedTypes.includes(event.target.files[0].type)) {
+        setInvalidImageMessage('Please upload a valid image file!')
+        return;
+      };
+      if (event.target.files[0].size > 1000000) {
```
Comment (jacoblb64) [inline app/src/components/task-completion/TaskUpload.tsx:84]: This is 1MB right? Image size has been varying, but this is probably too small -- most uploads are in the 2-5MB range, we'll have to set this higher at say, 10MB

**Review by jacoblb64 — CHANGES_REQUESTED** [review]
Looks good overall, but the cap is too low!

File: app/src/components/task-completion/TaskUpload.tsx (hunk 1 above)
Comment (sandaiiyahh) [reply app/src/components/task-completion/TaskUpload.tsx]: for HEIC photos, a preview is no longer shown. Is that ok? I did try uploading a file through my iPhone on an online code editor with <input> and it seems to convert live photos taken and camera roll images to jpeg/jpg/png

Comment (jacoblb64) [reply app/src/components/task-completion/TaskUpload.tsx]: Ah great! If it auto converts then no big deal, we can accept or not accept them, either way is fine by me!

**Review by jacoblb64 — APPROVED** [review]
Thanks for making the changes! Feel free to drop heic support if you think it's better

**Comment by sandaiiyahh** [comment]
> Thanks for making the changes! Feel free to drop heic support if you think it's better

I'll leave it in!

**Pull request merged**